app/services/dossier_filter_service.rb

```diff
@@ -90,6 +90,8 @@ def self.filtered_ids(dossiers, filters)
             dossiers.joins(:corrections).where(corrections: DossierCorrection.pending)
           elsif filtered_column.column == "state" && values.include?("en_construction")
             dossiers.where("dossiers.#{column} IN (?)", values).includes(:corrections).where.not(corrections: DossierCorrection.pending)
+          elsif filtered_column.type == :integer
+            dossiers.where("dossiers.#{column} IN (?)", values.map(&:to_i))
```

Comment:
to_i fallback sur 0 lorsqu ec'est pas un nombre, et on ferait donc une recherche sur 0, ce qui peut renvoyer d'autres résultats. C'est moins élégant mais on pourrait faire : 

```suggestion
            dossiers.where("dossiers.#{column} IN (?)", values.filter_map { Integer(_1) rescue nil })
```

Reply:
Bien vu, j'ai pris ta suggestion